snuba/utils/streams/backends/kafka.py

```diff
@@ -637,6 +643,25 @@ def build_kafka_consumer_configuration(
     }
 
 
+def build_kafka_producer_configuration(
+    broker_config: Mapping[str, Any],
+) -> Mapping[str, Any]:
```

Comment:
We are trying to fully type hint snuba. Which means we would prefer dataclasses with a well defined list of attributes and their types to move around the code base. This would make the code that depends on the configuration harder to break. 
So could you please declare a dataclass (like `KafkaBrokerConfiguration`) and return an instance here that is built from the settings dictionary ? (better to keep settings as simple dictionaries for now).

Basically like we did for clusters (https://github.com/getsentry/snuba/blob/master/snuba/clusters/cluster.py#L273). Those were not datatclasses but it does not apply here.

Reply:
I see your point, but the return value of `build_kafka_consumer_configuration`, `build_kafka_producer_configuration` and `build_admin_client_configuration` is directly passed to the confluent-kafka classes: Consumer, Producer and AdminClient.

I can make these function return a data class, only to immediately convert the data class back to a dictionary, when instantiating the library classes.

Would this be better than the current approach?

Reply:
Updated the code to reflect the above comment in https://github.com/getsentry/snuba/pull/1406/commits/778458fe183829d2dc80ef661c9c1a5eb0ba528e. I'm not really happy with how it looks. wdyt?